### PR TITLE
FM2-36: Implement Search to get all entries in every provider

### PR DESCRIPTION
**Pull request by Andu033**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-36

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**.

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java [hunk 1]
```diff
@@ -29,9 +29,9 @@
 	public static final String OPENMRS_FHIR_EXT_ADDRESS = OPENMRS_FHIR_EXT_PREFIX + "/address";
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java]: Please remove this from the PR

File: api/src/main/java/org/openmrs/module/fhir2/api/FhirPersonService.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
 import java.util.Date;
 
 public interface FhirPersonService {
+
+	Collection<Person> findPeople();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/FhirPersonService.java]: This should be `findAllPeople()`

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirPersonDao.java [hunk 3]
```diff
@@ -16,6 +16,8 @@
 import java.util.Date;
 
 public interface FhirPersonDao {
+
+	Collection<Person> findPeople();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirPersonDao.java]: `findAllPeople()`

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java [hunk 4]
```diff
@@ -41,7 +41,12 @@
 	@Inject
 	@Named("sessionFactory")
 	SessionFactory sessionFactory;
-	
+
+	@Override
+	public List<Patient> findAllPatients() {
+		return patientService.getAllPatients();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java]: Could we use the pattern `sessionFactory.getCurrentSession().createCriteria(...).list();` here?

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java [hunk 5]
```diff
@@ -28,11 +28,16 @@
...
-	
+
 	@Inject
 	@Named("sessionFactory")
 	SessionFactory sessionFactory;
-	
+
+	@Override
+	public Collection<Location> findAllLocations() {
+		return locationService.getAllLocations();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java]: Could we use the pattern `sessionFactory.getCurrentSession().createCriteria(...).list();` here?

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPractitionerDaoImpl.java [hunk 6]
```diff
@@ -35,7 +36,12 @@
 	@Inject
 	@Named("sessionFactory")
 	private SessionFactory sessionFactory;
-	
+
+	@Override
+	public List<Provider> findAllProviders() {
+		return providerService.getAllProviders();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPractitionerDaoImpl.java]: Could we use the pattern `sessionFactory.getCurrentSession().createCriteria(...).list();` here?

File: api/src/main/java/org/openmrs/module/fhir2/api/FhirPatientService.java [hunk 7]
```diff
@@ -9,13 +9,15 @@
  */
 package org.openmrs.module.fhir2.api;
 
-import java.util.Collection;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/FhirPatientService.java]: Please make sure your IDE is configured according to the [OpenMRS standard](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE). In particular, `java` and `javax` import should come before other imports (except static).

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java [hunk 8]
```diff
@@ -24,6 +19,19 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Collection;
+
+import static org.hibernate.criterion.Restrictions.eq;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java]: This should be at the top of the file

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java [hunk 9]
```diff
@@ -24,6 +19,19 @@
...
+
+/**
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java]: I don't think we need these remarks here

File: omod/src/main/java/org/openmrs/module/fhir2/providers/LocationFhirResourceProvider.java [hunk 10]
```diff
@@ -11,8 +11,8 @@
 
 import ca.uhn.fhir.rest.annotation.IdParam;
 import ca.uhn.fhir.rest.annotation.Read;
-import ca.uhn.fhir.rest.annotation.RequiredParam;
 import ca.uhn.fhir.rest.annotation.Search;
+import ca.uhn.fhir.rest.annotation.RequiredParam;
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/fhir2/providers/LocationFhirResourceProvider.java]: Why this change in import order?

File: omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java [hunk 11]
```diff
@@ -9,11 +9,6 @@
  */
 package org.openmrs.module.fhir2.web.servlet;
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java]: Please remove this from the PR

File: omod/src/main/resources/config.xml [hunk 12]
```diff
@@ -103,6 +103,5 @@
 		<defaultValue>5021b1a1-e7f6-44b4-ba02-da2f2bcf8718</defaultValue>
```
Comment (ibacher) [inline omod/src/main/resources/config.xml]: Please remove this file from the PR

**Comment by Andu033** [comment]
@ibacher  I did the requested changes. Should i write tests for the new endpoints?

File: api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConceptServiceImpl.java [hunk 13]
```diff
@@ -28,7 +29,12 @@
 	
 	@Inject
 	private FhirConceptDao dao;
-	
+
+	@Override
+	public Collection<Concept> getAllConcepts() {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConceptServiceImpl.java]: We don't currently need this

File: api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConceptSourceServiceImpl.java [hunk 14]
```diff
@@ -29,7 +29,12 @@
 	
 	@Inject
 	private FhirConceptSourceDao dao;
-	
+
+	@Override
+	public Collection<FhirConceptSource> getAllFhirConceptSources() {
+		return dao.getFhirConceptSources();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConceptSourceServiceImpl.java]: We don't currently need this

File: api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirObservationServiceImpl.java [hunk 15]
```diff
@@ -20,6 +20,9 @@
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.Collection;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirObservationServiceImpl.java]: `java` package imports should be above the others.

**Comment by Andu033** [comment]
@ibacher Thanks for the review. I did the requested changes.

File: api/src/main/java/org/openmrs/module/fhir2/api/FhirConceptService.java [hunk 16]
```diff
@@ -9,11 +9,14 @@
  */
```
[the remark this answers is not in the document]
Comment (ibacher) [reply api/src/main/java/org/openmrs/module/fhir2/api/FhirConceptService.java]: Can you eliminate this file from the PR?

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirServiceRequestDao.java [hunk 17]
```diff
@@ -11,9 +11,14 @@
 
 import javax.validation.constraints.NotNull;
 
+import org.hl7.fhir.r4.model.ServiceRequest;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirServiceRequestDao.java]: We don't need this import here

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImpl.java [hunk 18]
```diff
@@ -23,6 +23,7 @@
 import org.openmrs.api.PersonService;
 import org.openmrs.module.fhir2.api.dao.FhirPersonDao;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
```
[the remark this answers is not in the document]
Comment (ibacher) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImpl.java]: Please remove this import

**Comment by ibacher** [comment]
> @ibacher I did the requested changes. Should i write tests for the new endpoints?

@Andu033 Yes, it'd be great if you could write some tests for the new endpoints. You may need to re-check out this branch from your Github, since I forced some changes to get things to compile. To do so, in your local copy run the following commands:

```
git checkout master
git branch -D FM2-36
git fetch --all --prune
git checkout -b FM2-36
git pull origin FM2-36
```

**Comment by ibacher** [comment]
Closing this for now.
